mmv1/third_party/terraform/fwprovider/framework_provider_credentials_test.go

```diff
@@ -0,0 +1,278 @@
+package fwprovider_test
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func TestAccFwProvider_credentials(t *testing.T) {
```

Comment:
For comparison, the test cases in this PR's deletions are:

- `credentials can be configured as a path to a credentials JSON file`
- `configuring credentials as a path to a non-existent file results in an error`
- `credentials set in the config are not overridden by environment variables`
- `when credentials is unset in the config, environment variables are used: GOOGLE_CREDENTIALS used first`
- `when credentials is unset in the config, environment variables are used: GOOGLE_CLOUD_KEYFILE_JSON used second`
- `when credentials is unset in the config, environment variables are used: GCLOUD_KEYFILE_JSON used third`
- `when credentials is unset in the config (and access_token unset), GOOGLE_APPLICATION_CREDENTIALS is used...` (used after the ENVs above)
- `when credentials is set to an empty string in the config the value isn't ignored and results in an error`